src/Package/Impl/Sql/Extensions/SqlStringExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Linq;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.Sql {
+    internal static class SqlStringExtensions {
+        public static string ToSqlName(this string name, SqlQuoteType quoteType) {
```

Comment:
This should also escape the closing quote character if it occurs in the name (doubling does the trick for both quote and closing bracket).